packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -300,17 +301,18 @@ export class SchemaFactory<
 			public constructor(input: FlexTreeNode | unknown) {
 				super();
 				// Currently this just does validation. All other logic is in the subclass.
-				if (isFlexTreeNode(input)) {
-					assert(
-						getClassSchema(input.schema) === this.constructor,
-						0x83b /* building node with wrong schema */,
+				if (isFlexTreeNode(input) && getClassSchema(input.schema) !== this.constructor) {
```

Comment:
this case only happens when a flex tree node is provided. As this case is only triggered internally, and this is an invariant the internals should maintain, I think an assert was the correct choice here.

Reply:
Unlike the below error, it should not be possible to his this from misuse of the public API